server/translations.ts

```diff
@@ -60,7 +61,42 @@ function packageTranslationDifferences(translationDifferences) {
 }
 
 const _foundDocumentsCache = new Map();
+const sourceCommitCache = fs.existsSync("./source-commit.json")
+  ? new Map<string, number>(
+      Object.entries(
+        JSON.parse(fs.readFileSync("./source-commit.json", "utf8"))
+      )
+    )
+  : new Map<string, number>();
+const memCommitStore = new Map<string, string[]>();
+let memCommitStoreOldest = "HEAD";
 export async function findDocuments({ locale }) {
+  function checkCacheValidation(prevCache: Map<any, any>): void {
+    const contentHash = getRecentRepoHash(CONTENT_ROOT);
+    const translatedContentHash = getRecentRepoHash(CONTENT_TRANSLATED_ROOT);
+
+    function getRecentRepoHash(cwd: string): string {
+      return execSync("git rev-parse HEAD", { cwd }).toString().trimEnd();
+    }
+    function isValidCache(): boolean {
+      return (
+        prevCache.has(CONTENT_ROOT) &&
+        prevCache.has(CONTENT_TRANSLATED_ROOT) &&
+        prevCache.get(CONTENT_ROOT) === contentHash &&
+        prevCache.get(CONTENT_TRANSLATED_ROOT) === translatedContentHash
+      );
+    }
+
+    if (!isValidCache()) {
+      prevCache.clear();
+      prevCache.set(CONTENT_ROOT, contentHash);
+      prevCache.set(CONTENT_TRANSLATED_ROOT, translatedContentHash);
+      sourceCommitCache.clear();
```

Comment:
However, as written this cache check/invalidation leaves the `source-commit.json` file a bit superfluous, as the `_foundDocumentsCache` is always going to be empty, and therefore invalid, at startup, meaning `source-commit.json` is loaded above but then immediately invalidated here.

I say we've got a few options:
1. we think our speed up is enough to allow that ~15s wait (on my computer) every time we start the server, and don't cache anything to disk (removing the `source-commit.json` file)
2. do away with the `source-commit.json` cache and instead cache the entire `_foundDocumentsCache` to file
3. store the relevant hashes in `source-commit.json` so we can invalidate that separately

You have the experience using this dashboard, so I'll let you decide how irritating the various tradeoffs (long startup, not being able to invalidate the cache by simply restarting the server, etc.) would be and decide which option to take

Reply:
Reflected in the commit below. Details will be explained in the full comment.

- https://github.com/mdn/yari/pull/8338/commits/afcee51449565915cbb2faf67fe30ed1b05465ad